Выполнены условия сдачи задания на ревью

poofik33 opened the pull request.

poofik33 committed:
Added .gitignore file

poofik33 committed:
Add solutions of task files, tests for task and CMakeList

poofik33 committed:
Added changes in .gitignore for debug build

poofik33 committed:
Create .travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Correct .travis.yml

poofik33 committed:
Change minimal version cmake for v3.12

poofik33 committed:
Change minimal version CMake for 3.12

poofik33 committed:
added Travis build status

poofik33 committed:
added ls for check make results

poofik33 committed:
added valgrind test with gtest in Travis

poofik33 committed:
fix valgrind error

poofik33 committed:
Merge branch 'for_check' of https://github.com/poofik33/cpp_personal_task1 into for_check

poofik33 committed:
Added name in README.md

IlyaSaneev commented on post.c:
```diff
@@ -0,0 +1,348 @@
+    if (!posts) {
+        return false;
+    }
+    *posts = (post*)malloc(sizeof(post) * array_size);
```
стоит проверять указатель сразу после вызова маллос

IlyaSaneev commented on post.c:
```diff
@@ -0,0 +1,348 @@
+
+// функция создания массива постов через консольный ввод/вывод
+bool create(post **posts, size_t array_size) {
+    if (!posts) {
```
array_size == 0?

IlyaSaneev commented on post.h:
```diff
@@ -0,0 +1,75 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <time.h>
```
1.инкулды отсортировать
2. в post.h стоит поместить только те хидеры которые нужно для описание прототипов функций, все остальные перенести в .c

IlyaSaneev commented on post.h:
```diff
@@ -0,0 +1,75 @@
+    const post *p_post;
+};
+
+bool create(post **posts, size_t array_size);
```
create( - либо переименовать функцию либо написать коммент, что именно создает функция

IlyaSaneev commented on post.h:
```diff
@@ -0,0 +1,75 @@
+
+char* input_string();
+struct tm * input_date();
+int input_number();
```
методы к постам не относятся, вынести их в отдельные c/h файлы

IlyaSaneev commented on post.h:
```diff
@@ -0,0 +1,75 @@
+struct tm * input_date();
+int input_number();
+
+void delete_content(post *posts, size_t array_size);
```
delete_content -> delete_posts

IlyaSaneev commented on main.c:
```diff
@@ -0,0 +1,27 @@
+        free(popular_posts);
+    }
+    delete_content(posts, array_size);
+    return 0;
```
return EXIT_SUCCESS

IlyaSaneev commented on main.c:
```diff
@@ -0,0 +1,27 @@
+    printf("Введите количество постов\n");
+    size_t array_size = input_number();
+    bool success = create(&posts, array_size);
+    if (success) {
```
if (success) { ->
if (!success) { return EXIT_FAILURE; } 
тогда не будет лишнего блока

IlyaSaneev commented on post.c:
```diff
@@ -0,0 +1,348 @@
+        return false;
+    for (unsigned i = 0; i < array_size; i++) {
+        post *new_post = NULL;
+        bool success = create_post(&new_post);
```
может можно в create_post сразу нельзя передать &posts_array[i], тогда копировать память будет не нужно

IlyaSaneev commented on post.c:
```diff
@@ -0,0 +1,348 @@
+}
+
+// функция создания поста через консольный ввод/вывод
+bool create_post(post **new_post_p) {
```
декомпозируй метод

IlyaSaneev commented on tests_gtest.cpp:
```diff
@@ -0,0 +1,90 @@
+// Created by poofik on 12.10.2019.
+//
+#include <gtest/gtest.h>
+#include <unordered_set>
```
unordered_set ? :)

IlyaSaneev reviewed (commented):
1. cppcheck
2. разнести вызовы утилит санитайзеров, сборки и тестов на разные стадии пайплайна
3. посчитать покрытие

poofik33 committed:
Merge branch 'for_check' of https://github.com/poofik33/cpp_personal_task1 into for_check

poofik33 committed:
Fix review comments

- added checks for pointers after malloc
- added checks for array sizes
- added define for success exit and exit failure
- change function create_post to work with allocated memory
- decomposed function create_post, added functions for adding
  commentaries array and ratings array
- moved input functions in input.c/.h
- sorted include files
- renamed functions create and delete_content
- removed excess includes int gtest

poofik33 committed:
Add file test.txt for run with valgrind

poofik33 committed:
Update travis.yml

Added stages, cppcheck, memcheck main program

poofik33 committed:
Update .travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Fix travis.yml

poofik33 committed:
Fix travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Update .travis.yml

poofik33 committed:
Updated CMakeLists.txt for code coverage

poofik33 committed:
Update travis.yml for coverage

poofik33 committed:
Update travis.yml for code coverage with codecov

poofik33 committed:
Fix travis.yml

poofik33 committed:
Fix travis.yml

poofik33 committed:
Add badge code coverage

poofik33 committed:
Update README.md

poofik33 committed:
Fix code coverage badge

IlyaSaneev commented on .travis.yml:
```diff
@@ -0,0 +1,42 @@
+jobs:
+  include:    
+    - stage: "cppcheck"
+      script: cppcheck -j4 --inconclusive --enable=all --language=c main.c post.c post.h input.c input.h
```
хорошо бы прерывать пайплайн если cppcheck или другие санитайзеры завершились с ошибкой

IlyaSaneev commented on post.c:
```diff
@@ -0,0 +1,253 @@
+
+    bool success = create_comments_array(&new_post_p->comments, &new_post_p->comments_count, new_post_p->publication_date);
+    if (!success)
+        return false;
```
форматирование - слишком длинная строка

IlyaSaneev commented on post.c:
```diff
@@ -0,0 +1,253 @@
+
+    printf("Введите количество комментариев\n");
+    *array_size = input_number();
+    struct comment *comment_array = (struct comment *)malloc(sizeof(struct comment) * *array_size);
```
нет проверки

IlyaSaneev commented on post.h:
```diff
@@ -0,0 +1,57 @@
+//
+// Created by poofik on 12.10.2019.
+//
```
Это лишнее - обычно здест копирайт

IlyaSaneev reviewed (commented):
Можно мержить

poofik33 committed:
Update .gitignore

poofik33 committed:
Fix 2nd review comments in code files

poofik33 committed:
Add check for malloc error situation

poofik33 committed:
Update travis.yml check errors in sanitizers

poofik33 committed:
Merge branch 'master' into for_check
